developers/release.md

```diff
@@ -0,0 +1,348 @@
+
+Congratulations, you have successfully branched all of the repositories and prepared them for release. We recommend double checking each repository ensuring that the branch exists in the upstream repository, that the top of the commit stack in that branch includes the tag you created prior to branching (so that you have an initial tag point that represents the branch point) and that the tag itself exists. The last part of this step is to send to a follow up email letting the community know that everything has been branched.
+
+### Step 5. Pin katello-installer puppet modules for release
+
+On the katello-installer's KATELLO-X.Y branch, we need to pin the modules to a minor release.
+
+First, we take the pinned Foreman modules in the Puppetfile from the most current release, e.g. https://github.com/theforeman/foreman-installer/blob/1.9-stable/Puppetfile.  Overwrite the Foreman section in katello-installer's Puppetfile with these locked down versions.
+
+Next, we'll lock down the Katello modules. First, Check out all the puppet modules:
+
+```bash
+for module in candlepin certs capsule common elasticsearch gutterball katello crane service_wait pulp
+do
+  git clone git@github.com:Katello/puppet-${module}.git
+done
+```
+
+Review the modules for releasing - we should be keeping them up to date when making incompatible changes, but do any need bumps of the major or minor version? If so, do so now.
+
+Consider the changes that have been made to the module since the last release, do we need a new release?
+Use [blacksmith](https://github.com/maestrodev/puppet-blacksmith) to push to forge.
+
+```bash
+rake module:release
+```
+
+Now, lock down Katello versions to the appropriate minor, e.g. if you release `puppet-katello 0.2.0` or the latest is 0.2.0, then in the Puppetfile
+
+```
+mod 'katello',   '>= 0.2.0 < 0.3.0'
+```
+
+### Step 6: Configure Koji
+
+Note that to perform this step, Foreman must have generated it's Koji build targets for the targeted Foreman release. This is due to how Katello tags inherit from some Foreman tags. Coordinate with the Foreman release nanny on when they plan to do that step and wait to do this step until then. Koji needs to have configuration added specific to the tags that the release will occur under. The first step is to add new build roots and tags using the following script:
+
+```
+https://github.com/Katello/katello-misc/blob/master/rel-eng/koji_create_buildroots.sh
+```
+
+We recommend running this script in logical chunks instead of attempting to run the entire thing at once.
+
+### Step 7: Setup Mash Scripts
+
+In order to generate the release repositories on Koji, a mash script for the version being released needs to be added. If you do not have direct access to the koji.katello.org box, you can request access or ask someone who does to perform this step for you.
+
+Copy previous mash script to new one
+
+```
+cp /usr/local/bin/katello-mash-split-2.0.py /usr/local/bin/katello-mash-split-2.1.py
```

Comment:
i would recommend we copy from the nightly mash script, as any updates that occur to maintain nightly (such as adding new clients), would then propagate to the new release